packages/slate-editor/src/extensions/galleries/components/Gallery/GalleryTile.tsx

```diff
@@ -1,37 +1,93 @@
+import { GalleryPadding } from '@prezly/slate-types';
 import classNames from 'classnames';
 import type { CSSProperties } from 'react';
-import React from 'react';
+import React, { forwardRef } from 'react';
 
-import { ImageSizeWarning, ImageWithLoadingPlaceholder } from '#components';
+import { Button, ImageSizeWarning, ImageWithLoadingPlaceholder } from '#components';
+import { Crop, Delete } from '#icons';
 
 import styles from './GalleryTile.module.scss';
 
-interface Props {
+export interface Props {
+    caption?: string;
     className?: string;
-    url: string;
-    imageWidth: number;
+    clone?: boolean;
+    dragging?: boolean;
     imageHeight: number;
+    imageWidth: number;
+    insertPosition?: 'before' | 'after';
+    isInteractive?: boolean;
+    onCaptionChange: (caption: string) => void;
+    onCrop: () => void;
+    onDelete: () => void;
+    padding?: GalleryPadding;
     style?: CSSProperties;
+    url: string;
     withBorderRadius: boolean;
     withSizeWarning?: boolean;
+    withSmallButtons?: boolean;
 }
 
-export function GalleryTile({
-    className,
-    url,
-    imageWidth,
-    imageHeight,
-    style = {},
-    withBorderRadius,
-    withSizeWarning = false,
-}: Props) {
+export const GalleryTile = forwardRef<HTMLDivElement, Props>(function GalleryTile(
+    {
+        caption,
+        className,
+        clone = false,
+        dragging,
+        imageHeight,
+        imageWidth,
+        insertPosition,
+        isInteractive,
+        onCaptionChange,
+        onCrop,
+        onDelete,
+        padding,
+        style = {},
+        url,
+        withBorderRadius,
+        withSizeWarning = false,
+        withSmallButtons = false,
+        ...props
+    },
+    ref,
+) {
     return (
         <div
             className={classNames(styles.GalleryTile, className, {
                 [styles.withBorderRadius]: withBorderRadius,
+                [styles.interactive]: isInteractive,
+                [styles.dragging]: dragging,
+                [styles.insertBefore]: insertPosition === 'before',
+                [styles.insertAfter]: insertPosition === 'after',
+                [styles.withMediumPadding]: padding === GalleryPadding.M,
+                [styles.withLargePadding]: padding === GalleryPadding.L,
```

Comment:
I've noticed that if you drag quickly enough, the tiles first get the hover state and then immediately lose it. Not sure if it's possible to fix it.

[Screencast from 2024-02-23 19-48-42.webm](https://github.com/prezly/slate/assets/370680/18ffef8b-cfb3-4780-91cf-3d6aa3cf589f)

Reply:
I've tried fixing it with a timeout of 0, can you try it out? 7823d8411c2583e3251125722295849462ea78b3